docs/extending/developing-code/extending/case-studies/Undefined/skeletal.rst

```diff
@@ -0,0 +1,104 @@
+First Skeletal version
+=======================
+
+.. contents::
+   :depth: 1
+   :local:
+
+Armed with the information above we are now ready to draft the first version of this.
+
+Mathics built-in symbols are defined by writing a class which is based off of the class ``Builtin``.
+
+The docstring for this class lists
+
+* The common name for this, if a corresponding match can be found Wikipedia, use that name and link to it
+* links to the WMA reference we found before.
+* If there is a corresponding SymPy, mpmath, or SciPy function, then link to that
+* A HTML "definition list" to enclose the definition of each form that can appear
+* User-Oriented Examples
+
+``Undefined`` Title Line
+-------------------------
+
+Here, there is no Wikipedia common name, so we'll go with the Mathematica-like description "Undefined symbol or value". The WMA link is put in parenthesis after that.
+See the next case study :ref:`Adding Builtin Function ``KroneckerProduct``` for an example where we fill a Wikipedia and SymPy entry.
+
+To start out then we have:
+
+
+.. code-block:: python
+
+
+    class Undefined(Builtin):
+        """
+        Undefined symbol/value (<url>:WMA: https://reference.wolfram.com/language/ref/Undefined.html</url>)
+	...
+	"""
+
+The slightly unusual way we add a link with text is via the XML ``<url>`` tag.
+Also, due to current limitations in our homegrown documentation system, the title line has to appear as one unbroken line. To accommodate this, We have arrange ``pylint`` to ignore long lines.
+
+Note: please add the class ``Undefined`` in alphabetic order of class name. At the time of writing, ``Undefined`` comes after ``Pi`` and before ``Underflow``.
+
+``Undefined`` Symbol Definition Description
+--------------------------------------------
+
+Let us filling out the docstring further, the line above with ``...`` in it.
+
+After the title line, we get to the definition list part where we list the forms that this can appear. Here there is just one form ``Undefined``:
+
+.. code-block:: python
+
+    class Undefined(Builtin):
```

Comment:
Predefined instead Builtin?

Reply:
Predefined seems like Bultin except it adds one function that seems unnecessary. What the advantage of Predefined?